### PR TITLE
Added docs explaining multiple --restrict-to-path flags, with example…

### PR DESCRIPTION
**Pull request by enkore**
… (take 2)

### CODE REVIEW

File: docs/deployment.rst [hunk 1]
```diff
@@ -54,6 +54,11 @@ Restrictions
 Borg is instructed to restrict clients into their own paths:
 ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>``
 
+The client will be able to access any file or subdirectory inside of ``/home/backup/repos/<client fqdn>``
+but no other directories.  You can allow a client to access several directories by passing multiple
+`--restrict-to-path` flags, for instance: ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>/root --restrict-to-path /home/backup/repos/<client fqdn>/home``,
+or instead simply use `--restrict-to-path` once to restrict the client to ``/home/backup/repos/<client fqdn>/*``.
+
```
Comment (enkore) [inline docs/deployment.rst]: re https://github.com/borgbackup/borg/pull/1605#discussion_r79062680 

Probably ok, the usage docs explain the option, here the explanation in the "official example".

File: docs/deployment.rst [hunk 2]
```diff
@@ -54,6 +54,11 @@ Restrictions
 Borg is instructed to restrict clients into their own paths:
 ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>``
 
+The client will be able to access any file or subdirectory inside of ``/home/backup/repos/<client fqdn>``
+but no other directories.  You can allow a client to access several directories by passing multiple
+`--restrict-to-path` flags, for instance: ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>/root --restrict-to-path /home/backup/repos/<client fqdn>/home``,
```
Comment (ThomasWaldmann) [inline docs/deployment.rst]: maybe a different example should be given for the "multiple" case, not one that is better solved by pointing to the parent dir, as given in the first example.

File: docs/deployment.rst [hunk 3]
```diff
@@ -54,6 +54,12 @@ Restrictions
 Borg is instructed to restrict clients into their own paths:
 ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>``
 
+The client will be able to access any file or subdirectory inside of ``/home/backup/repos/<client fqdn>``
+but no other directories. You can allow a client to access several separate directories by passing multiple
+`--restrict-to-path` flags, for instance: ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>/root --restrict-to-path /home/backup/repos/<other client fqdn>``,
+which could make sense if multiple machines belong to one person which should then have access to all the
+backups of their machine.
```
Comment (ThomasWaldmann) [inline docs/deployment.rst]: ... of their machines.

File: docs/deployment.rst [hunk 4]
```diff
@@ -54,6 +54,12 @@ Restrictions
 Borg is instructed to restrict clients into their own paths:
 ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>``
 
+The client will be able to access any file or subdirectory inside of ``/home/backup/repos/<client fqdn>``
+but no other directories. You can allow a client to access several separate directories by passing multiple
+`--restrict-to-path` flags, for instance: ``borg serve --restrict-to-path /home/backup/repos/<client fqdn>/root --restrict-to-path /home/backup/repos/<other client fqdn>``,
```
Comment (ThomasWaldmann) [inline docs/deployment.rst]: the "/root" part is not needed any more for this example.

File: borg/archiver.py [hunk 5]
```diff
@@ -1011,7 +1011,9 @@ def build_parser(self, args=None, prog=None):
                                           help='start repository server process')
         subparser.set_defaults(func=self.do_serve)
         subparser.add_argument('--restrict-to-path', dest='restrict_to_paths', action='append',
-                               metavar='PATH', help='restrict repository access to PATH')
+                               metavar='PATH', help='restrict repository access to PATH. '
+                                                    'Can be specified multiple times to allow the client access to several directories. '
+                                                    'Access to all sub-directories is granted implicitly; PATH doesn\'t need to directly point to a repository.')
```
Comment (ThomasWaldmann) [inline borg/archiver.py]: good :)

File: docs/deployment.rst (hunk 4 above)
Comment (enkore) [inline docs/deployment.rst]: fixd

**Pull request merged**